verifier/src/main/kotlin/tech/figure/classification/asset/verifier/client/VerifierClient.kt

```diff
@@ -174,6 +175,19 @@ class VerifierClient(private val config: VerifierClientConfig) {
             EventIgnoredUnknownWasmEvent(event).send()
             return
         }
+        // If multiple asset classification smart contracts are instantiated and used in the same Provenance Blockchain
+        // environment, their events will all be detected by this client.  This check avoids attempting to process
+        // correctly-formatted contract events from a wholly different contract than the one registered in the configuration's
+        // ACClient instance
+        config.acClient.queryContractAddress().also { contractAddress ->
```

Comment:
I was going to ask if we should cache this value, but looks like the DefaultACQuerier already does, nice!